build/template_kql/common/extend_resource.kql

```diff
@@ -413,6 +453,8 @@ resources
     (Type == 'microsoft.sql/servers/databases'), "Not Applicable", 
     (Type == 'microsoft.documentdb/databaseaccounts'), iif(properties.backupPolicy.periodicModeProperties.backupStorageRedundancy == "Geo", "Healthy", "Unhealthy"),
     (Type == 'microsoft.dbformysql/servers'), iif(properties.storageProfile.geoRedundantBackup == "Enabled", "Healthy", "Unhealthy"),
+    (Type == 'microsoft.dbforpostgresql/servers'), iif(properties.storageProfile.geoRedundantBackup == "enabled", "healthy", "unhealthy"),
+    (Type == 'microsoft.dbforpostgresql/flexibleservers'), properties.backup.geoRedundantBackup,
```

Comment:
It's necessary that the resource is replicated or not, because PostgreSQL's replica cannot enable Backup.
![image](https://github.com/Azure/reliability-workbook/assets/3822284/161edd16-22a5-4e23-abfc-daca5ee2f432)

You can use the same code as like AutomaticFailover.

Reply:
Assuming you are meaning this code. 
`   type == 'microsoft.dbforpostgresql/flexibleservers', case(
        properties.highAvailability.mode == "Disabled" and properties.replicationRole != "GeoAsyncReplica" , "Not Configured",
        properties.replicationRole == "GeoAsyncReplica", "Not Applicable",
        properties.highAvailability.mode == "ZoneRedundant", "Zone Redundant",
        properties.highAvailability.mode == "SameZone", "Same Zone",
        properties.highAvailability.mode
    ),`